src/dos/drive_fat.cpp

```diff
@@ -1229,34 +1233,45 @@ bool fatDrive::FindNext(DOS_DTA &dta) {
 	return FindNextInternal(dta.GetDirIDCluster(), dta, &dummyClust);
 }
 
-bool fatDrive::GetFileAttr(char *name, Bit16u *attr) {
-	direntry fileEntry;
-	Bit32u dirClust, subEntry;
-	if(!getFileDirEntry(name, &fileEntry, &dirClust, &subEntry)) {
-		char dirName[DOS_NAMELENGTH_ASCII];
-		char pathName[11];
+bool fatDrive::GetFileAttr(char *name, Bit16u *attr)
+{
+	direntry fileEntry = {};
+	uint32_t dirClust, subEntry;
 
-		/* Can we even get the name of the directory itself? */
-		if(!getEntryName(name, &dirName[0])) return false;
-		convToDirFile(&dirName[0], &pathName[0]);
+	/* you CAN get file attr root directory */
+	if (*name == 0) {
+		*attr = DOS_ATTR_DIRECTORY;
+		return true;
+	}
 
-		/* Get parent directory starting cluster */
-		if(!getDirClustNum(name, &dirClust, true)) return false;
+	if (!getFileDirEntry(name, &fileEntry, &dirClust, &subEntry, true)) {
+		return false;
+	} else
+		*attr = fileEntry.attrib;
+	return true;
+}
 
-		/* Find directory entry in parent directory */
-		Bit32s fileidx = 2;
-		if (dirClust==0) fileidx = 0;	// root directory
-		Bit32s last_idx=0;
-		while(directoryBrowse(dirClust, &fileEntry, fileidx, last_idx)) {
-			if(memcmp(&fileEntry.entryname, &pathName[0], 11) == 0) {
-				*attr=fileEntry.attrib;
-				return true;
-			}
-			last_idx=fileidx;
-			fileidx++;
-		}
+bool fatDrive::SetFileAttr(const char *name, uint16_t attr)
+{
+	if (readonly) {
+		DOS_SetError(DOSERR_ACCESS_DENIED);
+		return false;
+	}
+	direntry fileEntry = {};
+	uint32_t dirClust, subEntry;
+
+	/* you cannot set file attr root directory (right?) */
+	if (*name == 0) {
+		DOS_SetError(DOSERR_ACCESS_DENIED);
 		return false;
-	} else *attr=fileEntry.attrib;
+	}
+
+	if (!getFileDirEntry(name, &fileEntry, &dirClust, &subEntry, true)) {
+		return false;
+	} else {
```

Comment:
Suggest moving these variables:
```
	direntry fileEntry = {};
	uint32_t dirClust, subEntry;
```
beneath the name check, so they're constructed just before we're written into by the `getFileDirEntry` function.

Reply:
(also suggest putting variables one-per-line, which are easier to see)

Reply:
Done.